cilium/CFP-42466-change-cilium-pod-ip-pool-crd.md

```diff
@@ -0,0 +1,198 @@
+type CIDREntry struct {
+    Allocator     cidralloc.CIDRAllocator
+    CIDR          string
+    Unallocatable bool
+}
+
+type cidrPool struct {
+    v4         []CIDREntry
+    v6         []CIDREntry
+    v4MaskSize int
+    v6MaskSize int
+}
+```
+The new CIDREntry type allows each CIDR to carry its own metadata, including the Unallocatable flag.
+
+### Allocator Unallocatable Logic Example
+
+```go
+// pkg/ipam/allocator/multipool/pool_allocator.go
+func (p *PoolAllocator) allocateCIDRs(targetNode, sourcePool string, family ipam.Family, toAllocate *big.Int) error {
+	zero := new(big.Int)
+	if toAllocate.Cmp(zero) <= 0 {
+		return nil // nothing to allocate
+	}
+  if p.isUnallocatable(sourcePool, cidr) {
+      p.logger.Warn(
+          "Skipping allocation from unallocatable CIDR",
+          logfields.SourcePool, sourcePool,
+          logfields.CIDR, cidr,
+      )
+      continue
+  }
+}
+
+func (p *PoolAllocator) isUnallocatable(poolName string, cidr netip.Prefix) bool {
+    pool, ok := p.pools[poolName]
+    if !ok {
+        return false
+    }
+
+    for _, entry := range pool.v4 { 
+        if entry.CIDR == cidr.String() && entry.Unallocatable {
+            return true
+        }
+    }
+    return false
+}
+```
+
+
+## Impacts / Key Questions
+
+
+### Impact 1 : CRD Version Promotion (`v2alpha1` → `v2`)
+The CiliumPodIPPool resource currently exists only in cilium.io/v2alpha1.Introducing per-CIDR configuration fields (e.g., unallocatable) requires changing the schema from a list of strings to a list of structured objects.This will likely require promoting the CRD to cilium.io/v2.
+
+This change affects backward compatibility and upgrade paths, as clusters using existing v2alpha1 resources might need migration logic.
+
+### Key Question: Should we promote the CRD to v2 directly or extend v2alpha1 first for testing?
+
+### Option 1: Promote directly to `v2`
+
+#### Pros
+- Simplifies long-term maintenance.
+- Avoids confusion between alpha and stable versions.
+
+#### Cons
+- Requires immediate migration handling and testing during the promotion process.
+- Existing users may need to update and validate their configurations right away, which increases short-term implementation overhead.
+
+### Option 2: Keep `v2alpha1` and introduce the new schema there first
+
+#### Pros
+
+- Allows iterative development and testing based on community feedback.
+- Reduces migration and compatibility risks before the API promotion.
+
+#### Cons
+- May cause confusion or compatibility issues for existing users already relying on v2alpha1.
+
+### Impact 2: Operator Behavior Change
+The Cilium Operator currently assumes that all CIDRs in the pool are allocatable.
+Adding an unallocatable: true field changes allocation logic and reconciliation behavior.
+
+This will affect IPAM logic — especially how nodes select pools and how the operator reports “available IPs.”
+There may be a need to modify the allocator to skip unallocatable ranges.
```

Comment:
I don't know what this section tries to say. Of course the operator needs to be changed, isn't that the whole point of the feature?